### PR TITLE
refactor: normalize imports in tests

### PR DESCRIPTION
**Pull request by ephys**
## Pull Request Checklist

_Please make sure to review and check all of these items:_

- [ ] Have you added new tests to prevent regressions?
- [x] Does `yarn test` or `yarn test-DIALECT` pass with this change (including linting)?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)?

### Description Of Change

This PR updates how Sequelize is imported from tests to reduce accessing `/lib` as much as possible for two reasons:

- We plan on removing the ability to import `/lib`
- We often have to update them as the TS migration continues, this will reduce that noise by accessing the public API as much as possible.

After this PR, `/lib` will only be referenced 42 times in tests (instead of hundred of times). The remaining exports are things that cannot be imported from the root package, such as dialects.

In the future, these exports will be imported through the `/_use-at-your-own-risk_/` package (inspired by https://github.com/eslint/eslint/blob/main/package.json#L13):

```
require('@sequelize/core/lib/dialects/abstract/query-generator');
// becomes
require('@sequelize/core/_use-at-your-own-risk_/dialects/abstract/query-generator');
```

This PR is part of a series of PR that reduce the size of https://github.com/sequelize/sequelize/pull/14280

### CODE REVIEW

File: test/unit/dialects/mariadb/errors.test.js [hunk 1]
```diff
@@ -5,7 +5,8 @@ const chai = require('chai');
 const expect = chai.expect;
 const Support = require('../../support');
 
-const Sequelize = Support.Sequelize;
+const { Sequelize } = require('@sequelize/core');
+
```
Comment (WikiRik) [inline test/unit/dialects/mariadb/errors.test.js:9]:
```suggestion
```

**Review by WikiRik — APPROVED** [review]
Assuming that tests pass it looks good to me. Suggestions are optional

File: test/integration/dialects/postgres/regressions.test.js [hunk 2]
```diff
@@ -5,7 +5,8 @@ const chai = require('chai');
 const expect = chai.expect;
 const Support = require('../../support');
 
-const Sequelize = Support.Sequelize;
+const { Sequelize } = require('@sequelize/core');
+
```
[the remark this answers is not in the document]
Comment (ephys) [reply test/integration/dialects/postgres/regressions.test.js:9]: eslint will force a empty line after requires but the one before this one can be removed :)

[the remark this answers is not in the document]
Comment (ephys) [reply test/integration/dialects/postgres/regressions.test.js:9]: I think I'll just leave it as it, eslint will ensure import groups are clean once these files are migrated to TS :)

**Pull request merged**